# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1376959410 -7200
#      Tue Aug 20 02:43:30 2013 +0200
# Node ID ba817fecfbf083706a33803d87165d9974304344
# Parent  3ac729a302d3e588f905a53634d5575ef950095a
-Fix [FS#5715] (r25729): Don't use negative unsigneds (MSVC C4146).

diff --git a/src/widget.cpp b/src/widget.cpp
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -1248,18 +1248,18 @@ void NWidgetHorizontal::AssignSizePositi
 
 	/* Third loop: Compute position and call the child. */
 	uint position = rtl ? this->current_x : 0; // Place to put next child relative to origin of the container.
 	NWidgetBase *child_wid = this->head;
 	while (child_wid != NULL) {
 		uint child_width = child_wid->current_x;
-		uint child_x = x + position + (rtl ? -child_width - child_wid->padding_left : child_wid->padding_left);
+		uint child_x = x + (rtl ? position - child_width - child_wid->padding_left : position + child_wid->padding_left);
 		uint child_y = y + child_wid->padding_top;
 
 		child_wid->AssignSizePosition(sizing, child_x, child_y, child_width, child_wid->current_y, rtl);
 		uint padded_child_width = child_width + child_wid->padding_right + child_wid->padding_left;
-		position += rtl ? -padded_child_width : padded_child_width;
+		position = rtl ? position - padded_child_width : position + padded_child_width;
 
 		child_wid = child_wid->next;
 	}
 }
 
 /** Horizontal left-to-right container widget. */
